escape_roomba/run_bot.py

```diff
@@ -0,0 +1,78 @@
+"""Main Discord bot program.
+
+Called by the poetry-generated 'run_bot' script which invokes main() below
+(see pyproject.yaml).
+"""
+
+import logging
+import os
+import signal
+
+import discord
+
+import escape_roomba.context
+
+
+def main():
+    """Main entry point from 'run_bot' wrapper script (see pyproject.yaml)."""
+
+    signal.signal(signal.SIGINT, signal.SIG_DFL)  # Sane ^C behavior.
+    logging.basicConfig(
+        level=logging.INFO,
+        format='%(asctime)s %(name)s [%(levelname)s] %(message)s',
+        datefmt='%m-%d %H:%M:%S')
+    for name in logging.root.manager.loggerDict:  # Suppress library chatter.
+        logging.getLogger(name).setLevel(logging.WARNING)
+
+    try:
+        bot_token = os.environ['ESCAPE_ROOMBA_BOT_TOKEN']
+    except KeyError as e:
+        logging.critical(f'*** No ${e.args[0]}! See README.md.')
+        return 1
+
+    context = escape_roomba.context.Context(discord.Client())
+    status_logger = StatusLogger(context)
+    context.client.run(bot_token)
+
+
+class StatusLogger:
+    """Logs notable server/connection events."""
+
+    def __init__(self, context):
+        self.context = context
+        context.add_listener_methods(self)
+
+    async def on_connect(self):
+        logging.info('Connected to Discord:\n'
+                     f'    {self.context.client.ws.gateway}')
```

Comment:
Nit: Is this whitespace intentional? (Maybe it looks good in practice, not sure.)

Reply:
It is! Indents in the log output.